sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -351,10 +351,24 @@ object PartitioningUtils {
    */
   def getPathFragment(spec: TablePartitionSpec, partitionSchema: StructType): String = {
     partitionSchema.map { field =>
-      escapePathName(field.name) + "=" + getPartitionValueString(spec(field.name))
+      escapePathName(field.name) + "=" +
+        getPartitionValueString(
+          removeLeadingZerosFromNumberTypePartition(spec(field.name), field.dataType))
     }.mkString("/")
   }
 
+  def removeLeadingZerosFromNumberTypePartition(value: String, dataType: DataType): String = {
+    dataType match {
+      case ByteType => value.toByte.toString
```

Comment:
Would it not be sufficient to call .toLong.toString for any integer type? might be simpler.
Also, would float/double ever be used as a partition key?

Reply:
> Would it not be sufficient to call .toLong.toString for any integer type? might be simpler.

Yes, i will modify it.
> Also, would float/double ever be used as a partition key?

Fload/Double type is rarely used as a partition key, but i think we should handle them